apm-sniffer/bytebuddy-patch/src/main/java/net/bytebuddy/agent/builder/NativeMethodStrategySupport.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package net.bytebuddy.agent.builder;
+
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+
+import java.lang.reflect.Field;
+
+/**
+ * Inject custom NativeMethodStrategy to AgentBuilder
+ */
+public class NativeMethodStrategySupport {
+    private static final String PREFIX = "sw_origin$";
+    private static ILog LOGGER = LogManager.getLogger(NativeMethodStrategySupport.class);
+
+    public static void inject(AgentBuilder agentBuilder) {
+        Class<? extends AgentBuilder> clazz = agentBuilder.getClass();
+        if (clazz != AgentBuilder.Default.class) {
+            throw new IllegalStateException("Only accept original AgentBuilder instance but not a wrapper instance: " + clazz.getName());
+        }
+        try {
+            Field nativeMethodStrategyField = clazz.getDeclaredField("nativeMethodStrategy");
+            nativeMethodStrategyField.setAccessible(true);
+            nativeMethodStrategyField.set(agentBuilder, new SWNativeMethodStrategy(PREFIX));
```

Comment:
These seem a hijacking and relying on upstream(bytebuddy) doesn't change the internal implementation.
Considering bytebuddy is always very friendly to our community, you are better to submit a proposal/request to bytebuddy, and discuss the official API to do so.

Reply:
OK, That's what I am thinking.